server/datastore/mysql/aggregated_stats.go

```diff
@@ -122,7 +122,8 @@ func (ds *Datastore) CalculateAggregatedPerfStatsPercentiles(ctx context.Context
 	// NOTE: this function gets called for query and scheduled_query, so the id
 	// refers to a query/scheduled_query id, and it never computes "global"
 	// stats. For that reason, we always set global_stats=0.
-	_, err = tx.ExecContext(ctx,
+	_, err = ds.writer(ctx).ExecContext(
```

Comment:
Won't there be data races with multiple Fleet instances receiving live query results from hosts for the same query?

Reply:
Yes, and we currently have that problem. If 2 live queries are kicked off, it is possible that only 1 stats will be written to scheduled_query_stats. So, we could lose 1 stats measurement.

I think this is fine because:
- this is a corner case -- why would 2 identical live queries be kicked off simultaneously?
- query stats are not a critical feature -- missing 1 measurement is ok
- we prioritize master DB performance, which is a bottleneck for some customers

Reply:
Ah I just realized only one Fleet instance processes results, so we are g2g here.

Reply:
Good question. Good answer.